src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -285,28 +286,22 @@ private void Poll()
                 {
                     // no outstanding requests so we don't expect any messages back, but we should anyway
                     // drive the KafkaConsumer by polling to handle partition events etc.
-                    // TODO: Make consumer pause-resume calls work
-                    // _consumer.Pause(currentAssignment);
+                    _consumer.Pause(currentAssignment);
                     var message = _consumer.Consume(TimeSpan.FromMilliseconds(1));
                     if (message != null)
                     {
-                        // TODO: Change this once pause-resume will work
-                        // While we are not pausing partitions, we need to shift position back when accidently consumed message we are not interested in
-                        _consumer.Seek(message.TopicPartitionOffset);
-                        // throw new IllegalActorStateException($"Got unexpected Kafka message: {message.ToJson()}");
+                        throw new IllegalActorStateException($"Got unexpected Kafka message: {message.ToJson()}");
                     }
                 }
                 else
                 {
                     // resume partitions to fetch
                     IImmutableSet<TopicPartition> partitionsToFetch = _requests.Values.SelectMany(v => v.Topics).ToImmutableHashSet();
-                    /*
-                    // TODO: Make consumer pause-resume calls work
                     var resumeThese = currentAssignment.Where(partitionsToFetch.Contains).ToList();
                     var pauseThese = currentAssignment.Except(resumeThese).ToList();
                     _consumer.Pause(pauseThese);
-                    _consumer.Resume(resumeThese);
-                    */
+                    _consumer.Resume(resumeThese.Except(_resumedPartitions));
+                    _resumedPartitions = _resumedPartitions.Union(resumeThese);
```

Comment:
@IgorFedchenko do we have any specs at the moment which cover pause / resume behavior? Given the issue with the underlying Confluent driver, we should have a regression check in-place just to ensure that if they patch the driver in a future release it doesn't break the work-around we've added in here. Is that something you can add or does it exist already?

Reply:
@Aaronontheweb First of all, before this workaround current tests were failing (that is why all pause/resume code was commented out) - so if they will break it that much tests will fail again.
But I understand what are you talking about, seems like we need some special tests to ensure pause/resume partitions working correctly.
While current tests make use of this pause/resume calls, currently there is never any partitions to pause actually - because all subscribed partitions are the ones that stage is interested in. The real pause/resume usage will start when I will add `ExternalSingleSource` stage with `ExternalSingleSourceLogic` class - because that kind of stage allows sharing this consuming actor between stages. And that is where multiple stages will requests for different partitions consumed, and some of them will be paused (when there is no demand on them) - and some of them should be resumed/consumed.
So that is great idea to make sure we have some regression tests here - I've just created issue #55 to not forget about that. I think I will close it once implement that `ExternalSingleSource` with it's tests.

Reply:
Ok, sounds good - I'll merge this in for now then and we can add a future PR to address that going forward.